app/services/submission/plant_trial_finalizer.rb

```diff
@@ -35,10 +35,80 @@ def create_new_trait_descriptors
 
   def create_scoring
     trait_mapping = submission.content.step04.trait_mapping
+    trait_scores = submission.content.step04.trait_scores || {}
+    accessions = submission.content.step04.accessions
+    replicate_numbers = submission.content.step04.replicate_numbers || {}
+    design_factors = submission.content.step04.design_factors || {}
+    design_factor_names = submission.content.step04.design_factor_names || []
+    lines_or_varieties = submission.content.step04.lines_or_varieties || {}
+
+    @new_plant_scoring_units = trait_scores.map do |plant_id, scores|
+      rollback(2) unless accessions[plant_id] && accessions[plant_id]['plant_accession'] && accessions[plant_id]['originating_organisation']
```

Comment:
Yes, this "should never happen(tm)". I simply didn't want the situation when our broken code allows submission of some incorrect data. Should I rather simply throw an uncaught exception (500)?